src/main/java/com/czertainly/core/service/impl/AuthServiceImpl.java

```diff
@@ -28,8 +33,11 @@ public class AuthServiceImpl implements AuthService {
 
     @Override
     @AuditLogged(originator = ObjectType.FE, affected = ObjectType.ACCESS, operation = OperationType.REQUEST)
-    public UserDto getAuthProfile() throws NotFoundException {
-        return userManagementApiClient.getUserProfile().getData().getUser();
+    public UserDto getAuthProfile() throws NotFoundException, JsonProcessingException {
+        CzertainlyUserDetails userDetails = (CzertainlyUserDetails) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
```

Comment:
In case of support of anonymous user, handle case of not authenticated user.
If no authentication or principal is present, throw exception with message of Unknown or anonymous user

Reply:
log and Message updated